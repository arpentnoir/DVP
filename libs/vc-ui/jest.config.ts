/* eslint-disable */
export default {
  displayName: 'vc-ui',
  preset: '../../jest.preset.js',
  setupFiles: ['./jest.setup.ts'],
  transform: {
    '^.+\\.[tj]sx?$': 'babel-jest',
  },
  moduleFileExtensions: ['ts', 'tsx', 'js', 'jsx'],
  coverageDirectory: '../../coverage/libs/vc-ui',
  moduleNameMapper: {
    '^react-pdf': 'react-pdf/dist/umd/entry.jest',
  },
};
