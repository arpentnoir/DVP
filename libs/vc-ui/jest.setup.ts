import 'jest-canvas-mock';
import '@testing-library/jest-dom';

const mockClipboard = {
  writeText: jest.fn(),
};

(global.navigator.clipboard as any) = mockClipboard;
