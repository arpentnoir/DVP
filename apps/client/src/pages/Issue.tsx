import { Form, FormOption, Text, VCOptions } from '@dvp/vc-ui';
import { Grid, Paper } from '@mui/material';
import { useNavigate } from 'react-router-dom';
import { BaseLayout } from '../layouts';
import {
  schema as partialCooSchema,
  schemaUI as partialCooSchemaUi,
} from './CreateForm/AANZFTACOOPartialSchema';
import {
  schema as fullCooSchema,
  schemaUI as fullCooSchemaUi,
} from './CreateForm/AANZFTACOOSchema';

export const Issue = () => {
  // TODO: Request forms from the backend.
  const forms = [
    {
      id: '001',
      name: 'AANZFTA-COO',
      fullForm: { schema: fullCooSchema, uiSchema: fullCooSchemaUi },
      partialForm: { schema: partialCooSchema, uiSchema: partialCooSchemaUi },
    },
  ];

  const navigate = useNavigate();

  const goToForm = (form: Form, credentialType: any) => {
    navigate('/form', {
      state: {
        form,
        credentialType,
      },
    });
  };

  const handleFormSelected = (selectedForm: FormOption) => {
    const { partialForm, fullForm, formType, credentialType } = selectedForm;

    goToForm(
      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
      formType === 'partial' ? (partialForm as Form) : (fullForm as Form),
      credentialType
    );
  };

  return (
    <BaseLayout title="Form Select">
      <Text variant="h4" fontWeight="bold" paddingBottom="24px">
        Issue document
      </Text>
      <Text>
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis
        molestie, dictum est a, mattis tellus. Sed dignissim, metus nec
        fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus
        elit sed risus. Maecenas eget condimentum velit, sit amet feugiat
        lectus. Class aptent taciti sociosqu ad litora torquent per conubia
        nostra, per inceptos himenaeos. Praesent auctor purus luctus enim
        egestas, ac scelerisque ante pulvinar. Donec ut rhoncus ex. Suspendisse
        ac rhoncus nisl, eu tempor urna. Curabitur vel bibendum lorem. Morbi
        convallis convallis diam sit amet lacinia. Aliquam in elementum tellus.
      </Text>
      <Grid container>
        <Grid item display={'flex'} justifyContent={'center'} width={'100%'}>
          <Paper
            sx={{
              width: '100%',
              maxWidth: '38rem',
              padding: { xs: '20px', md: '56px' },
            }}
          >
            <VCOptions forms={forms} onFormSelected={handleFormSelected} />
          </Paper>
        </Grid>
      </Grid>
    </BaseLayout>
  );
};
