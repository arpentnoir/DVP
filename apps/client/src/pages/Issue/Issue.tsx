import { Form, FormOption, Text, VCOptions } from '@dvp/vc-ui';
import { Grid, Paper } from '@mui/material';
import { useNavigate } from 'react-router-dom';
import { BaseLayout } from '../../layouts';
import {
  schema as partialCooSchema,
  schemaUI as partialCooSchemaUi,
} from '../CreateForm/AANZFTACOOPartialSchema';
import {
  schema as fullCooSchema,
  schemaUI as fullCooSchemaUi,
} from '../CreateForm/AANZFTACOOSchema';

export const Issue = () => {
  // TODO: Request forms from the backend.
  const forms = [
    {
      id: '001',
      name: 'AANZFTACoO',
      displayName: 'AANZFTA-COO',
      fullForm: { schema: fullCooSchema, uiSchema: fullCooSchemaUi },
      partialForm: { schema: partialCooSchema, uiSchema: partialCooSchemaUi },
    },
  ];

  const navigate = useNavigate();

  const goToForm = (
    form: Form,
    credentialType: any,
    name: string,
    formType?: string
  ) => {
    navigate('/form', {
      state: {
        form,
        credentialType,
        formName: name,
        formType,
      },
    });
  };

  const handleFormSelected = (selectedForm: FormOption) => {
    const { partialForm, fullForm, formType, credentialType, name } =
      selectedForm;

    goToForm(
      // eslint-disable-next-line @typescript-eslint/no-unnecessary-type-assertion
      formType === 'partial' ? (partialForm as Form) : (fullForm as Form),
      credentialType,
      name,
      formType
    );
  };

  return (
    <BaseLayout title="Form Select">
      <Text
        variant="h1"
        fontWeight="bold"
        paddingBottom="24px"
        data-testid="issue-header"
      >
        Issue document
      </Text>
      <Text>
        Issue your Document as a Verifiable Credential. Select the options for
        your document below.
      </Text>
      <Grid container>
        <Grid item display={'flex'} justifyContent={'center'} width={'100%'}>
          <Paper
            sx={{
              width: '100%',
              maxWidth: '38rem',
              padding: { xs: '20px', md: '56px' },
            }}
          >
            <VCOptions forms={forms} onFormSelected={handleFormSelected} />
          </Paper>
        </Grid>
      </Grid>
    </BaseLayout>
  );
};
