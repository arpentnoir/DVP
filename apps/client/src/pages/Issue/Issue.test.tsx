import { render } from '@dvp/vc-ui';
import { Issue } from './Issue';

describe('Issue', () => {
  it('should render correctly', () => {
    const { baseElement } = render(<Issue />);

    expect(baseElement).toMatchSnapshot();
  });
});
