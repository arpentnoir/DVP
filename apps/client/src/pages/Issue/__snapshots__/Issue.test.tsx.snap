// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Issue should render correctly 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-disableGutters css-1y9y4eh-MuiContainer-root"
    >
      <h1
        class="MuiTypography-root MuiTypography-h1 css-1ne5dp9-MuiTypography-root"
        data-testid="issue-header"
        tabindex="0"
      >
        Issue document
      </h1>
      <p
        class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
        tabindex="0"
      >
        Lorem ipsum dolor sit amet, consectetur adipiscing elit. Etiam eu turpis molestie, dictum est a, mattis tellus. Sed dignissim, metus nec fringilla accumsan, risus sem sollicitudin lacus, ut interdum tellus elit sed risus. Maecenas eget condimentum velit, sit amet feugiat lectus. Class aptent taciti sociosqu ad litora torquent per conubia nostra, per inceptos himenaeos. Praesent auctor purus luctus enim egestas, ac scelerisque ante pulvinar. Donec ut rhoncus ex. Suspendisse ac rhoncus nisl, eu tempor urna. Curabitur vel bibendum lorem. Morbi convallis convallis diam sit amet lacinia. Aliquam in elementum tellus.
      </p>
      <div
        class="MuiGrid-root MuiGrid-container css-11lq3yg-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-n54w8s-MuiGrid-root"
        >
          <div
            class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 css-1xv4rem-MuiPaper-root"
          >
            <div
              class="MuiBox-root css-0"
            >
              <h5
                class="MuiTypography-root MuiTypography-h5 css-1sv2wdh-MuiTypography-root"
                tabindex="0"
              >
                Verifiable Credential Options
              </h5>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-1lof2qr-MuiFormControl-root"
              >
                <label
                  aria-label="Select a document to issue. "
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="select-input-label-id"
                  id="form-select-label"
                >
                  Document Type
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  data-testid="form-select-document-type"
                  role="listbox"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="form-select-label"
                    class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    id="select-input-label-id"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Document Type
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-1lof2qr-MuiFormControl-root"
              >
                <label
                  aria-label="Select a form type "
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="select-input-label-id-form-type"
                  id="form-select-label-form-type"
                >
                  Form Type
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  data-testid="form-select-form-type"
                  role="listbox"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="form-select-label-form-type"
                    class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    id="select-input-label-id-form-type"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Form Type
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiFormControl-root MuiFormControl-fullWidth css-1lof2qr-MuiFormControl-root"
              >
                <label
                  aria-label="Select a credential type "
                  class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-colorPrimary MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                  data-shrink="false"
                  for="select-input-label-id-credential-type"
                  id="form-select-label-credential-type"
                >
                  Credential Type
                </label>
                <div
                  class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-colorPrimary MuiInputBase-formControl  css-1yk1gt9-MuiInputBase-root-MuiOutlinedInput-root-MuiSelect-root"
                  data-testid="form-select-credential-type"
                  role="listbox"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    aria-labelledby="form-select-label-credential-type"
                    class="MuiSelect-select MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    role="button"
                    tabindex="0"
                  >
                    <span
                      class="notranslate"
                    >
                      ​
                    </span>
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    id="select-input-label-id-credential-type"
                    tabindex="-1"
                    value=""
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-yjsfm1"
                    >
                      <span>
                        Credential Type
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
              <div
                class="MuiBox-root css-1oknf09"
              >
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-1h2qilq-MuiButtonBase-root-MuiButton-root"
                  data-testid="button:Cancel"
                  tabindex="0"
                  type="button"
                >
                  <div
                    class="css-doi3ij-MuiStack-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      data-testid="button:Cancel:label"
                      tabindex="0"
                    >
                      Cancel
                    </p>
                  </div>
                  <span
                    class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                  />
                </button>
                <button
                  class="MuiButtonBase-root MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium Mui-disabled MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium css-1yt9wte-MuiButtonBase-root-MuiButton-root"
                  data-testid="button:Next"
                  disabled=""
                  tabindex="-1"
                  type="button"
                >
                  <div
                    class="css-doi3ij-MuiStack-root"
                  >
                    <p
                      class="MuiTypography-root MuiTypography-body2 css-e784if-MuiTypography-root"
                      data-testid="button:Next:label"
                      tabindex="0"
                    >
                      Next
                    </p>
                  </div>
                </button>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
